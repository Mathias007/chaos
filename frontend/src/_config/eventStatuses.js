const eventStatuses = {
    articles: {
        ARTICLE_SUCCESSFULLY_LOADED: "ARTICLE_SUCCESSFULLY_LOADED",
        ARTICLE_NOT_FOUND: "ARTICLE_NOT_FOUND",
        ARTS_LIST_LOADED: "ARTS_LIST_LOADED",
        ARTS_LIST_NOT_FOUND: "ARTS_LIST_NOT_FOUND",
        ARTICLE_ADDED: "ARTICLE_ADDED",
        ARTICLE_ADDING_FAILED: "ARTICLE_ADDING_FAILED",
        ARTICLE_SUCCESFULLY_EDITED: "ARTICLE_SUCCESFULLY_EDITED",
        ARTICLE_EDITING_FAILED: "ARTICLE_EDITING_FAILED",
        ARTICLE_SUCCESFULLY_DELETED: "ARTICLE_SUCCESFULLY_DELETED",
        ARTICLE_DELETING_FAILED: "ARTICLE_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    },
    auth: {
        PASSWORD_TOO_SHORT: "PASSWORD_TOO_SHORT",
        PASSWORD_INCORRECT: "PASSWORD_INCORRECT",
        PASSWORDS_NOT_MATCH: "PASSWORDS_NOT_MATCH",
        USER_LOADED: "USER_LOADED",
        LOGIN_SUCCESSFUL: "LOGIN_SUCCESSFUL",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        LOGIN_FAILED: "LOGIN_FAILED",
        LOGOUT_SUCCESSFUL: "LOGOUT_SUCCESSFUL",
        RESET_STATUS: "RESET_STATUS"
    },
    news: {
        NEWS_SUCCESSFULLY_LOADED: "NEWS_SUCCESSFULLY_LOADED",
        NEWS_NOT_FOUND: "NEWS_NOT_FOUND",
        NEWS_LIST_LOADED: "NEWS_LIST_LOADED",
        NEWS_LIST_NOT_FOUND: "NEWS_LIST_NOT_FOUND",
        NEWS_ADDED: "NEWS_ADDED",
        NEWS_ADDING_FAILED: "NEWS_ADDING_FAILED",
        NEWS_SUCCESFULLY_EDITED: "NEWS_SUCCESFULLY_EDITED",
        NEWS_EDITING_FAILED: "NEWS_EDITING_FAILED",
        NEWS_SUCCESFULLY_DELETED: "NEWS_SUCCESFULLY_DELETED",
        NEWS_DELETING_FAILED: "NEWS_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    },
    knights: {
        KAM_BUILDINGS_LOADED: "KAM_BUILDINGS_LOADED",
        KAM_BUILDINGS_NOT_FOUND: "KAM_BUILDINGS_NOT_FOUND",
        KAM_UNITS_LOADED: "KAM_UNITS_LOADED",
        KAM_UNITS_NOT_FOUND: "KAM_UNITS_NOT_FOUND",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR"
    },
    tzar: {
        TZAR_NATIONS_LOADED: "TZAR_NATIONS_LOADED",
        TZAR_NATIONS_NOT_FOUND: "TZAR_NATIONS_NOT_FOUND",
        TZAR_UNITS_LOADED: "TZAR_UNITS_LOADED",
        TZAR_UNITS_NOT_FOUND: "TZAR_UNITS_NOT_FOUND",
        TZAR_SPELLS_LOADED: "TZAR_SPELLS_LOADED",
        TZAR_SPELLS_NOT_FOUND: "TZAR_SPELLS_NOT_FOUND",
        TZAR_TECHNOLOGIES_LOADED: "TZAR_TECHNOLOGIES_LOADED",
        TZAR_TECHNOLOGIES_NOT_FOUND: "TZAR_TECHNOLOGIES_NOT_FOUND",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR"
    },
    users: {
        USER_SUCCESSFULLY_LOADED: "USER_SUCCESSFULLY_LOADED",
        USER_NOT_FOUND: "USER_NOT_FOUND",
        USERS_LIST_LOADED: "USERS_LIST_LOADED",
        USERS_LIST_NOT_FOUND: "USERS_LIST_NOT_FOUND",
        USER_ADDED: "USER_ADDED",
        USER_ADDING_FAILED: "USER_ADDING_FAILED",
        USER_SUCCESFULLY_EDITED: "USER_SUCCESFULLY_EDITED",
        USER_EDITING_FAILED: "USER_EDITING_FAILED",
        USER_SUCCESFULLY_DELETED: "USER_SUCCESFULLY_DELETED",
        USER_DELETING_FAILED: "USER_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    }
};

export default eventStatuses;
