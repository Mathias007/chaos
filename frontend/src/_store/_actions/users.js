import {
    getAllUsers,
    getSingleUser,
    addUser,
    editUser,
    deleteUser
} from "./services/UsersService";

import eventStatuses from "../../_config/eventStatuses";

const {
    USERS_LIST_LOADED,
    USERS_LIST_NOT_FOUND,
    USER_SUCCESSFULLY_LOADED,
    USER_NOT_FOUND,
    USER_ADDED,
    USER_ADDING_FAILED,
    USER_SUCCESFULLY_EDITED,
    USER_EDITING_FAILED,
    USER_SUCCESFULLY_DELETED,
    USER_DELETING_FAILED,
    AUTHENTICATION_ERROR
} = eventStatuses.users;

const { USER_LOADED } = eventStatuses.auth;

export const showUsersList = () => {
    return (dispatch, getState) => {
        const dispatchUsersListLoaded = function(response) {
            dispatch({
                type: USERS_LIST_LOADED,
                data: response.data,
                status: response.status
            });
        };

        const dispatchUsersAuthError = function(response) {
            dispatch({
                type: AUTHENTICATION_ERROR,
                data: response.data,
                status: response.status
            });
            throw response.data;
        };

        const dispatchUsersNotFound = function(response) {
            dispatch({
                type: USERS_LIST_NOT_FOUND,
                data: response.data,
                status: response.status
            });
        };

        return getAllUsers(
            dispatchUsersListLoaded,
            dispatchUsersAuthError,
            dispatchUsersNotFound
        );
    };
};

export const showUserProfile = id => {
    return (dispatch, getState) => {
        const dispatchSingleUserLoaded = function(response) {
            dispatch({
                type: USER_SUCCESSFULLY_LOADED,
                data: response.data,
                status: response.status
            });
        };

        const dispatchUserAuthError = function(response) {
            dispatch({
                type: AUTHENTICATION_ERROR,
                data: response.data,
                status: response.status
            });
            throw response.data;
        };

        const dispatchUserNotFound = function(response) {
            dispatch({
                type: USER_NOT_FOUND,
                data: response.data,
                status: response.status
            });
            throw response.data;
        };

        return getSingleUser(
            id,
            dispatchSingleUserLoaded,
            dispatchUserAuthError,
            dispatchUserNotFound
        );
    };
};

export const register = (name, email, password, role, date, remember) => {
    return (dispatch, getState) => {
        const dispatchRegistrationSuccessful = function(response) {
            dispatch({
                type: USER_ADDED,
                data: response.data,
                name: name,
                remember,
                status: response.status
            });
            dispatch({
                type: USER_LOADED,
                name,
                remember
            });

            return response.data, name, remember;
        };

        const dispatchRegistrationError = function(response) {
            dispatch({
                type: AUTHENTICATION_ERROR,
                data: response.data,
                status: response.status
            });
            throw response.data;
        };

        const dispatchRegistrationFailed = function(response) {
            dispatch({
                type: USER_ADDING_FAILED,
                data: response.data,
                status: response.status
            });
            throw response.data;
        };

        return addUser(
            name,
            email,
            password,
            role,
            date,
            dispatchRegistrationSuccessful,
            dispatchRegistrationError,
            dispatchRegistrationFailed
        );
    };
};

export const editSelectedUser = (id, name, email, password, role, date) => {
    return (dispatch, getState) => {
        const dispatchUserEdited = function(response) {
            dispatch({
                type: USER_SUCCESFULLY_EDITED,
                data: response.data,
                status: response.status
            });
            return response.data;
        };

        const dispatchUserAuthError = function(response) {
            dispatch({
                type: AUTHENTICATION_ERROR,
                data: response.data,
                status: response.status
            });
            throw response.data;
        };

        const dispatchUserEditingFailed = function(response) {
            dispatch({
                type: USER_EDITING_FAILED,
                data: response.data,
                status: response.status
            });
            throw response.data;
        };

        return editUser(
            id,
            name,
            email,
            password,
            role,
            date,
            dispatchUserEdited,
            dispatchUserAuthError,
            dispatchUserEditingFailed
        );
    };
};

export const deleteSelectedUser = id => {
    return (dispatch, getState) => {
        const dispatchUserDeleted = function(response) {
            dispatch({
                type: USER_SUCCESFULLY_DELETED,
                data: response.data,
                status: response.status
            });
            return response.data;
        };

        const dispatchUserAuthError = function(response) {
            dispatch({
                type: AUTHENTICATION_ERROR,
                data: response.data,
                status: response.status
            });
            throw response.data;
        };

        const dispatchUserDeletingFailed = function(response) {
            dispatch({
                type: USER_DELETING_FAILED,
                data: response.data,
                status: response.status
            });
            throw response.data;
        };

        return deleteUser(
            id,
            dispatchUserDeleted,
            dispatchUserAuthError,
            dispatchUserDeletingFailed
        );
    };
};
