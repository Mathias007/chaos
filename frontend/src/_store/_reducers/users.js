import eventStatuses from "../../_config/eventStatuses";

const {
    USER_SUCCESSFULLY_LOADED,
    USER_NOT_FOUND,
    USERS_LIST_LOADED,
    USERS_LIST_NOT_FOUND,
    USER_ADDED,
    AUTHENTICATION_ERROR,
    USER_ADDING_FAILED,
    USER_SUCCESFULLY_EDITED,
    USER_EDITING_FAILED,
    USER_SUCCESFULLY_DELETED,
    USER_DELETING_FAILED
} = eventStatuses.users;

export const initialState = {
    usersList: [],
    selectedUser: {},
    errorMessage: ""
};

export default function users(state = initialState, action) {
    switch (action.type) {
        case USERS_LIST_LOADED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                usersList: action.data.users
            };

        case USERS_LIST_NOT_FOUND:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case USER_SUCCESSFULLY_LOADED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                selectedUser: action.data.selectedUser
            };

        case USER_NOT_FOUND:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case USER_ADDED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case AUTHENTICATION_ERROR:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: "Something went wrong."
            };

        case USER_ADDING_FAILED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case USER_SUCCESFULLY_EDITED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                selectedUser: action.data.selectedUser,
                errorMessage: action.data.message
            };

        case USER_EDITING_FAILED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case USER_SUCCESFULLY_DELETED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case USER_DELETING_FAILED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        default:
            return state;
    }
}
