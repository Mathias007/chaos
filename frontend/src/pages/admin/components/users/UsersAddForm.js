import React, { Component } from "react";
import { connect } from "react-redux";

import {
    passwordPattern,
    siteRoles
} from "../../../../_config/globalContentVariables";
import { users } from "../../../../_store/_actions";
import linksPaths from "../../../../_config/linksPaths";
import serverStatuses from "../../../../_config/serverStatuses";
import styles from "../../../../styles/styles";

import SingleFormElement from "../../../components/universal/SingleFormElement";
import ButtonComponent from "../../../components/universal/ButtonComponent";
import ErrorMessageComponent from "../../../components/universal/ErrorMessageComponent";
import Success from "../../../components/errors/Success";

import { Form, Layout } from "antd";
const { Content } = Layout;

const { USERS } = linksPaths;
const { STATUS_OK } = serverStatuses;

class UsersAddForm extends Component {
    handleSubmit = e => {
        e.preventDefault();

        const { validateFields, resetFields } = this.props.form;

        validateFields((err, values) => {
            if (!err) {
                console.log("Received values of form: ", values);
                this.props.register(
                    values.username,
                    values.email,
                    values.password,
                    values.role,
                    values.date
                );
                resetFields();
            }
        });
    };

    render() {
        if (this.props.status === STATUS_OK)
            return (
                <Success
                    message={this.props.errorMessage}
                    continueFunction={this.props.cleanServerStatus}
                    cancelLink={USERS.MAIN}
                    description="Czy chcesz kontynuować pracę?"
                />
            );
        else {
            const { getFieldDecorator } = this.props.form;
            return (
                <Content style={styles.content}>
                    <Form onSubmit={this.handleSubmit} id="add-user-form">
                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="username"
                            inputIcon="user"
                            label="Nazwa użytkownika"
                            message="Określ unikalną nazwę użytkownika!"
                            placeholder="Podaj nazwę użytkownika"
                            required
                            Tooltip="Nazwą użytkownika jest ciąg znaków identyfikujący internautę w serwisie."
                        />
                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="email"
                            fieldType="email"
                            inputIcon="mail"
                            label="Zawartość artykułu"
                            message="Podaj poprawny adres e-mail!"
                            placeholder="Podaj adres e-mail"
                            required
                            Tooltip="Adres e-mail powiązany z kontem."
                        />
                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="password"
                            fieldType="password"
                            inputIcon="key"
                            label="Hasło"
                            message="Wpisz swoje hasło!"
                            minLength={8}
                            pattern={passwordPattern}
                            placeholder="Wpisz hasło użytkownika."
                            required
                            Tooltip="Hasło powinno składać się z conajmniej 8 znaków, zawierać literę oraz cyfrę."
                        />
                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="role"
                            fieldType="select"
                            initialValue="USER"
                            options={siteRoles}
                            label="Rola w serwisie"
                            message="Wpisz swoje hasło!"
                            required={false}
                        />
                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="date"
                            fieldType="date"
                            label="Data dodania lub utworzenia (pole nieobowiązkowe)"
                            inputIcon="calendar"
                            placeholder="Data utworzenia (domyślnie wygenerowana zostanie aktualna)"
                            required={false}
                        />
                        <ButtonComponent
                            htmlType="submit"
                            icon="user-add"
                            text="Dodaj użytkownika"
                            type="primary"
                        />
                        <ErrorMessageComponent
                            status={this.props.status}
                            errorMessage={this.props.errorMessage}
                        />
                    </Form>
                </Content>
            );
        }
    }
}

const mapStateToProps = state => {
    return {
        errorMessage: state.users.errorMessage,
        status: state.users.status
    };
};

const mapDispatchToProps = dispatch => {
    return {
        register: (username, email, password, role, date) => {
            return dispatch(
                users.register(username, email, password, role, date)
            );
        },
        cleanServerStatus: () => {
            return dispatch(users.cleanServerStatus());
        }
    };
};

UsersAddForm = connect(mapStateToProps, mapDispatchToProps)(UsersAddForm);

export default Form.create()(UsersAddForm);
