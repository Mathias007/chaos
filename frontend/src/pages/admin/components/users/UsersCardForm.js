import React, { Component } from "react";
import { connect } from "react-redux";

import { users } from "../../../../_store/_actions";
import {
    passwordPattern,
    siteRoles,
} from "../../../../_config/globalContentVariables";
import linksPaths from "../../../../_config/linksPaths";
import serverStatuses from "../../../../_config/serverStatuses";

import SingleFormElement from "../../../components/universal/SingleFormElement";
import ButtonComponent from "../../../components/universal/ButtonComponent";
import ErrorMessageComponent from "../../../components/universal/ErrorMessageComponent";
import Success from "../../../components/errors/Success";

import { Form } from "antd";

const { GENERAL, USERS } = linksPaths;
const { STATUS_OK } = serverStatuses;

class UsersCardForm extends Component {
    handleSubmit = (values) => {
        let modificationDate = new Date();

        console.log("Received values of form: ", values);
        this.props.editSelectedUser(
            this.props.idParam,
            values.username,
            values.email,
            values.password,
            values.role,
            modificationDate
        );
    };

    render() {
        if (this.props.status === STATUS_OK)
            return (
                <Success
                    message={this.props.errorMessage}
                    continueFunction={this.props.cleanServerStatus}
                    cancelLink={GENERAL.INDEX}
                    description="Czy chcesz wrócić do edycji profilu?"
                />
            );
        else {
            const { name, email, role } = this.props.selectedUser;
            return (
                <Form onFinish={this.handleSubmit} id="edit-user-form">
                    <SingleFormElement
                        fieldName="username"
                        initialValue={name}
                        inputIcon="user"
                        label="Nazwa użytkownika"
                        message="Określ unikalną nazwę użytkownika!"
                        placeholder="Podaj nazwę użytkownika"
                        required
                        Tooltip="Nazwą użytkownika jest ciąg znaków identyfikujący internautę w serwisie."
                    />

                    <SingleFormElement
                        fieldName="email"
                        fieldType="email"
                        initialValue={email}
                        inputIcon="mail"
                        label="Zawartość artykułu"
                        message="Podaj poprawny adres e-mail!"
                        placeholder="Podaj adres e-mail"
                        required
                        Tooltip="Adres e-mail powiązany z kontem."
                    />

                    <SingleFormElement
                        fieldName="password"
                        fieldType="password"
                        inputIcon="key"
                        label="Hasło"
                        message="Wpisz swoje hasło!"
                        minLength={8}
                        pattern={passwordPattern}
                        placeholder="Wpisz hasło użytkownika."
                        required={false}
                        Tooltip="Hasło powinno składać się z conajmniej 8 znaków, zawierać literę oraz cyfrę."
                    />

                    <SingleFormElement
                        fieldName="role"
                        fieldType="select"
                        initialValue={role}
                        options={siteRoles}
                        label="Rola w serwisie"
                        message="Wpisz swoje hasło!"
                        required={false}
                    />

                    <ButtonComponent
                        composition="double"
                        cancelLink={USERS.MAIN}
                        cancelText="Zrezygnuj"
                        htmlType="submit"
                        icon="user"
                        text="Edytuj profil"
                        type="primary"
                    />
                    <ErrorMessageComponent
                        status={this.props.status}
                        errorMessage={this.props.errorMessage}
                    />
                </Form>
            );
        }
    }
}

const mapStateToProps = (state) => {
    return {
        errorMessage: state.users.errorMessage,
        selectedUser: state.users.selectedUser,
        status: state.users.status,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        editSelectedUser: (
            username,
            email,
            password,
            role,
            modificationDate
        ) => {
            return dispatch(
                users.editSelectedUser(
                    username,
                    email,
                    password,
                    role,
                    modificationDate
                )
            );
        },
        cleanServerStatus: () => {
            return dispatch(users.cleanServerStatus());
        },
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(UsersCardForm);
