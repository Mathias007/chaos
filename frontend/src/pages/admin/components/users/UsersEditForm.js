import React, { Component } from "react";
import { connect } from "react-redux";

import { users } from "../../../_store/_actions";
import {
    passwordPattern,
    siteRoles
} from "../../../_config/globalContentVariables";
import linksPaths from "../../../_config/linksPaths";
import serverStatuses from "../../../_config/serverStatuses";
import styles from "../../../styles/styles";

import SingleFormElement from "../../components/SingleFormElement";
import ButtonComponent from "../../components/ButtonComponent";
import ErrorMessageComponent from "../../components/ErrorMessageComponent";
import Success from "../../errors/Success";

import { Form, Layout } from "antd";
const { Content } = Layout;

const { USERS } = linksPaths;
const { STATUS_OK } = serverStatuses;

class UserEditForm extends Component {
    handleSubmit = e => {
        e.preventDefault();

        let modificationDate = new Date();
        const { validateFields } = this.props.form;

        validateFields((err, values) => {
            if (!err) {
                console.log("Received values of form: ", values);
                this.props.editSelectedUser(
                    this.props.idParam,
                    values.username,
                    values.email,
                    values.password,
                    values.role,
                    modificationDate
                );
            }
        });
    };

    render() {
        if (this.props.status === STATUS_OK)
            return (
                <Success
                    message={this.props.errorMessage}
                    continueFunction={this.props.cleanServerStatus}
                    cancelLink={USERS.MAIN}
                    description="Czy chcesz kontynuować pracę?"
                />
            );
        else {
            const { getFieldDecorator } = this.props.form;
            const { selectedUser } = this.props;
            return (
                <Content style={styles.content}>
                    <Form onSubmit={this.handleSubmit} id="edit-user-form">
                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="username"
                            initialValue={selectedUser.name}
                            inputIcon="user"
                            label="Nazwa użytkownika"
                            message="Określ unikalną nazwę użytkownika!"
                            placeholder="Podaj nazwę użytkownika"
                            required
                            Tooltip="Nazwą użytkownika jest ciąg znaków identyfikujący internautę w serwisie."
                        />

                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="email"
                            fieldType="email"
                            initialValue={selectedUser.email}
                            inputIcon="mail"
                            label="Zawartość artykułu"
                            message="Podaj poprawny adres e-mail!"
                            placeholder="Podaj adres e-mail"
                            required
                            Tooltip="Adres e-mail powiązany z kontem."
                        />

                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="password"
                            fieldType="password"
                            inputIcon="key"
                            label="Hasło"
                            message="Wpisz swoje hasło!"
                            minLength={8}
                            pattern={passwordPattern}
                            placeholder="Wpisz hasło użytkownika."
                            required={false}
                            Tooltip="Hasło powinno składać się z conajmniej 8 znaków, zawierać literę oraz cyfrę."
                        />

                        <SingleFormElement
                            getFieldDecorator={getFieldDecorator}
                            fieldName="role"
                            fieldType="select"
                            initialValue={selectedUser.role}
                            options={siteRoles}
                            label="Rola w serwisie"
                            message="Wpisz swoje hasło!"
                            required={false}
                        />

                        <ButtonComponent
                            composition="double"
                            cancelLink={USERS.MAIN}
                            cancelText="Zrezygnuj"
                            htmlType="submit"
                            icon="user"
                            text="Edytuj użytkownika"
                            type="primary"
                        />
                        <ErrorMessageComponent
                            status={this.props.status}
                            errorMessage={this.props.errorMessage}
                        />
                    </Form>
                </Content>
            );
        }
    }
}

const mapStateToProps = state => {
    return {
        errorMessage: state.users.errorMessage,
        selectedUser: state.users.selectedUser,
        status: state.users.status
    };
};

const mapDispatchToProps = dispatch => {
    return {
        editSelectedUser: (
            username,
            email,
            password,
            role,
            modificationDate
        ) => {
            return dispatch(
                users.editSelectedUser(
                    username,
                    email,
                    password,
                    role,
                    modificationDate
                )
            );
        },
        cleanServerStatus: () => {
            return dispatch(users.cleanServerStatus());
        }
    };
};

UserEditForm = connect(mapStateToProps, mapDispatchToProps)(UserEditForm);

export default Form.create()(UserEditForm);
