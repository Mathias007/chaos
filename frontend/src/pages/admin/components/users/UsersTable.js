import React, { Component } from "react";
import { connect } from "react-redux";

import { users } from "../../../../_store/_actions";
import styles from "../../../../styles/styles";

import { usersColumnsStructure } from "../_helpers/usersColumnsStructure";

import { Layout, Table } from "antd";
const { Content } = Layout;

class UsersTable extends Component {
    state = {
        tableColumns: [],
        tableData: []
    };

    componentDidMount() {
        this.props.showUsersList();
    }

    componentDidUpdate(prevProps) {
        if (this.props.usersList.length !== prevProps.usersList.length) {
            this.props.showUsersList();
        }
    }

    renderUsers() {
        let { usersList } = this.props;

        if (usersList) {
            return usersList.map((singleUser, index) => {
                const { _id, name, email, role, date } = singleUser;

                return {
                    number: index,
                    key: _id,
                    avatar: name,
                    username: name,
                    email,
                    role,
                    date,
                    options: _id
                };
            });
        }
    }

    render() {
        let { tableColumns, tableData } = this.state;

        tableData = this.renderUsers();
        tableColumns = usersColumnsStructure;

        return (
            <Content style={styles.content}>
                <Table dataSource={tableData} columns={tableColumns} />
            </Content>
        );
    }
}

const mapStateToProps = state => {
    return {
        usersList: state.users.usersList
    };
};

const mapDispatchToProps = dispatch => {
    return {
        showUsersList: () => dispatch(users.showUsersList())
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(UsersTable);
