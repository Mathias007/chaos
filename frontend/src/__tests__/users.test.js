import eventStatuses from "../_config/eventStatuses";
import users from "../_store/_reducers/users";

const {
    USER_SUCCESSFULLY_LOADED,
    USER_NOT_FOUND,
    USERS_LIST_LOADED,
    USERS_LIST_NOT_FOUND,
    USER_ADDED,
    USER_ADDING_FAILED,
    USER_SUCCESFULLY_EDITED,
    USER_EDITING_FAILED,
    USER_SUCCESFULLY_DELETED,
    USER_DELETING_FAILED,
    AUTHENTICATION_ERROR
} = eventStatuses.users;

const initialState = {
    usersList: [],
    selectedUser: {},
    errorMessage: ""
};

describe("test users reducer", () => {
    const usersList = {
        message: "Pozytywna wiadomość z serwera!",
        data: [
            {
                _id: "1111-2222-3333-4444",
                name: "tester",
                email: "test@mail.com",
                password: "Terror2137",
                role: "TESTER",
                date: "01-01-2020"
            }
        ]
    };

    const singleUser = {
        message: "Okuratna wiadomość z serwera!",
        data: {
            _id: "1111-2222-3333-4444",
            name: "tester",
            email: "test@mail.com",
            password: "Terror2137",
            role: "TESTER",
            date: "01-01-2020"
        }
    };

    const message = "Ważna wiadomość z serwera!";

    it("should return the initial state", () => {
        expect(users(undefined, {})).toEqual(initialState);
    });

    it("should load users list", () => {
        const action = { type: USERS_LIST_LOADED, data: usersList.data };
        const expectedState = {
            ...initialState,
            ...action.data,
            usersList: action.data.usersList
        };
        expect(users(initialState, action)).toEqual(expectedState);
    });

    it("should get message about users list's not finding", () => {
        const action = { type: USERS_LIST_NOT_FOUND, data: message };
        const expectedState = {
            ...initialState,
            ...action.data,
            errorMessage: action.data.message
        };
        expect(users(initialState, action)).toEqual(expectedState);
    });

    it("should load a single user", () => {
        const action = {
            type: USER_SUCCESSFULLY_LOADED,
            data: singleUser.data
        };
        const expectedState = {
            ...initialState,
            ...action.data,
            selectedUser: action.data.singleUser
        };
        expect(users(initialState, action)).toEqual(expectedState);
    });

    it("should get message about single users' not finding", () => {
        const action = { type: USER_NOT_FOUND, data: message };
        const expectedState = {
            ...initialState,
            ...action.data,
            errorMessage: action.data.message
        };
        expect(users(initialState, action)).toEqual(expectedState);
    });

    it("should get message about adding user to database", () => {
        const action = { type: USER_ADDED, data: message };
        const expectedState = {
            ...initialState,
            ...action.data,
            errorMessage: action.data.message
        };
        expect(users(initialState, action)).toEqual(expectedState);
    });

    it("should get message about user's adding fail", () => {
        const action = { type: USER_ADDING_FAILED, data: message };
        const expectedState = {
            ...initialState,
            ...action.data,
            errorMessage: action.data.message
        };
        expect(users(initialState, action)).toEqual(expectedState);
    });

    it("should get message about modifying user in database", () => {
        const action = {
            type: USER_SUCCESFULLY_EDITED,
            data: { singleUser }
        };
        const expectedState = {
            ...initialState,
            ...action.data,
            selectedUser: action.data.data,
            errorMessage: action.data.message
        };
        expect(users(initialState, action)).toEqual(expectedState);
    });

    it("should get message about user's modifying fail", () => {
        const action = { type: USER_EDITING_FAILED, data: message };
        const expectedState = {
            ...initialState,
            ...action.data,
            errorMessage: action.data.message
        };
        expect(users(initialState, action)).toEqual(expectedState);
    });

    it("should get message about removing user from database", () => {
        const action = { type: USER_SUCCESFULLY_DELETED, data: message };
        const expectedState = {
            ...initialState,
            ...action.data,
            errorMessage: action.data.message
        };
        expect(users(initialState, action)).toEqual(expectedState);
    });

    it("should get message about user's removing fail", () => {
        const action = { type: USER_DELETING_FAILED, data: message };
        const expectedState = {
            ...initialState,
            ...action.data,
            errorMessage: action.data.message
        };
        expect(users(initialState, action)).toEqual(expectedState);
    });

    it("should get message about a problem with authentication", () => {
        const action = { type: AUTHENTICATION_ERROR, data: message };
        const expectedState = {
            ...initialState,
            ...action.data,
            errorMessage: action.data.message
        };
        expect(users(initialState, action)).toEqual(expectedState);
    });
});
