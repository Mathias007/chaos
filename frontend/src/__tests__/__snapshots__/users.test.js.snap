// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test users reducer should get message about a problem with authentication 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "selectedUser": Object {},
  "usersList": Array [],
}
`;

exports[`test users reducer should get message about adding user to database 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "selectedUser": Object {},
  "usersList": Array [],
}
`;

exports[`test users reducer should get message about modifying user in database 1`] = `
Object {
  "errorMessage": "Okuratna wiadomość z serwera!",
  "message": "Okuratna wiadomość z serwera!",
  "selectedUser": Object {
    "_id": "1111-2222-3333-4444",
    "date": "01-01-2020",
    "email": "test@mail.com",
    "name": "tester",
    "password": "Terror2137",
    "role": "TESTER",
  },
  "usersList": Array [],
}
`;

exports[`test users reducer should get message about removing user from database 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "selectedUser": Object {},
  "usersList": Array [],
}
`;

exports[`test users reducer should get message about single users' not finding 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "selectedUser": Object {},
  "usersList": Array [],
}
`;

exports[`test users reducer should get message about user's adding fail 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "selectedUser": Object {},
  "usersList": Array [],
}
`;

exports[`test users reducer should get message about user's modifying fail 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "selectedUser": Object {},
  "usersList": Array [],
}
`;

exports[`test users reducer should get message about user's removing fail 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "selectedUser": Object {},
  "usersList": Array [],
}
`;

exports[`test users reducer should get message about users list's not finding 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "selectedUser": Object {},
  "usersList": Array [],
}
`;

exports[`test users reducer should load a single user 1`] = `
Object {
  "errorMessage": "",
  "message": "Okuratna wiadomość z serwera!",
  "selectedUser": Object {
    "_id": "1111-2222-3333-4444",
    "date": "01-01-2020",
    "email": "test@mail.com",
    "name": "tester",
    "password": "Terror2137",
    "role": "TESTER",
  },
  "usersList": Array [],
}
`;

exports[`test users reducer should load users list 1`] = `
Object {
  "errorMessage": "",
  "message": "Pozytywna wiadomość z serwera!",
  "selectedUser": Object {},
  "users": Array [
    Object {
      "_id": "1111-2222-3333-4444",
      "date": "01-01-2020",
      "email": "test@mail.com",
      "name": "tester",
      "password": "Terror2137",
      "role": "TESTER",
    },
  ],
  "usersList": Array [
    Object {
      "_id": "1111-2222-3333-4444",
      "date": "01-01-2020",
      "email": "test@mail.com",
      "name": "tester",
      "password": "Terror2137",
      "role": "TESTER",
    },
  ],
}
`;
